integration/run_test.go

```diff
@@ -520,3 +521,152 @@ func TestRunInNamespace(t *testing.T) {
 		}
 	}
 }
+
+func TestRunWithIdMap(t *testing.T) {
+	type checker struct {
+		path        string
+		expectedUID string
+		expectedGID string
+	}
+
+	imageName := rabbitmqImage
+
+	baseSnapshotDir := "/var/lib/soci-snapshotter-grpc/snapshotter/snapshots"
+	baseRuntimeDir := "/run/containerd/io.containerd.runtime.v2.task/default"
+
+	uidPath := "/etc/subuid"
+	gidPath := "/etc/subgid"
+
+	dummyuser := "dummy-user"
+	dummygroup := "dummy-group"
+
+	modes := []struct {
+		name           string
+		indexBuilderFn func(sh *shell.Shell, src imageInfo, opts ...indexBuildOption) string
+	}{
+		{
+			name: "with only FUSE layers",
+			indexBuilderFn: func(sh *shell.Shell, src imageInfo, opts ...indexBuildOption) string {
+				opts = append(opts, withMinLayerSize(0))
+				return buildIndex(sh, src, opts...)
+			},
+		},
+		{
+			name: "with mixed layers",
+			indexBuilderFn: func(sh *shell.Shell, src imageInfo, opts ...indexBuildOption) string {
+				return buildIndex(sh, src, opts...)
+			},
+		},
+		{
+			name: "with no SOCI index",
+			indexBuilderFn: func(sh *shell.Shell, src imageInfo, opts ...indexBuildOption) string {
+				return ""
+			},
+		},
+	}
+
+	tests := []struct {
+		name           string
+		subUIDContents string
+		subGIDContents string
+		checkFiles     []checker
+	}{
+		{
+			name:           "with one set of substitutions",
+			subUIDContents: fmt.Sprintf("%s:12345:1001", dummyuser),
+			subGIDContents: fmt.Sprintf("%s:12345:1001", dummyuser),
+			checkFiles: []checker{
+				{
+					path:        "/usr/bin/sh",
+					expectedUID: "12345",
+					expectedGID: "12345",
+				},
+			},
+		},
+		{
+			name:           "with multiple substitutions",
+			subUIDContents: fmt.Sprintf("%s:12345:1000\n%s:22222:1", dummyuser, dummyuser),
```

Comment:
this test doesnt seem to be correct, as even if we dont do multiple uid and gid it will show the same result?

Reply:
What do you mean exactly?

Reply:
Test has been updated to use a different image with multiple user permissions inside.